Sources/Momento/TopicClient.swift

```diff
@@ -160,6 +168,39 @@ public class TopicClient: TopicClientProtocol {
      ```
      */
     public func subscribe(cacheName: String, topicName: String) async -> TopicSubscribeResponse {
+        return await self.doSubscribe(cacheName: cacheName, topicName: topicName, resumeAtTopicSequenceNumber: nil, resumeAtTopicSequencePage: nil)
+    }
+
+    /**
+     Subscribe to a topic. The returned value can be used to iterate over newly published messages on the topic.
+     - Parameters:
+        - cacheName: name of the cache containing the topic
+          topicName: name of the topic
+          resumeAtTopicSequenceNumber: sequence number to resume at. Use nil or 0 to start at the latest messages.
+          resumeAtTopicSequenceNumber: page number to resume at. Use nil or 0 to start at the latest messages.
+     - Returns: TopicSubscribeResponse representing the result of the subscribe operation.
+
+     Pattern matching can be used to operate on the appropriate subtype.
+     ```swift
+      switch subscribeResponse {
+      case .error(let err):
+          print("Error: \(err)")
+      case .subscription(let sub):
+          for try await item in sub.stream {...}
+      }
+     ```
+     */
+    public func subscribe(
+        cacheName: String, topicName: String, resumeAtTopicSequenceNumber: UInt64?, resumeAtTopicSequencePage: UInt64?
+    ) async -> TopicSubscribeResponse {
```

Comment:
is it possible to use a [default parameter](https://www.hackingwithswift.com/sixty/5/6/default-parameters) value in the existing signature so we avoid breaking the existing API but can stick to only one code path for subscribe?

e.g.
```
public func subscribe(
        cacheName: String, 
        topicName: String, 
        resumeAtTopicSequenceNumber: UInt64? = nil, 
        resumeAtTopicSequencePage: UInt64? = nil
    ) async -> TopicSubscribeResponse
```

Reply:
Let me play with that, thanks!

Reply:
The default parameter approach described in that link doesn't conform to a protocol like our `subscribe` method does. Because protocols can't specify default values, we end up in a situation where all 4 of the parameters must be supplied to the call despite the defaults. I can work around that by using a protocol extension to handle the default arguments, but this obscures the fact that the arguments are actually optional.

The practical effect of all this is that the IDE will only ever suggest a method signature with all 4 arguments supplied. Users can choose to remove the `resume*` arguments from the call, but they would have no way of knowing they *can* do that. Because the 99% case is to omit those arguments, and given that the additional code for the protocol extension is actually less easy to understand than the original implementation, I think we should stick with exposing two signatures for `subscribe`.